### PR TITLE
Add Integration tests for FSx Lustre Backup feature

### PR DESCRIPTION
**Pull request by siddharthsalot**
**Please See** [Git Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions)

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/storage/test_fsx_lustre.py [hunk 1]
```diff
@@ -71,6 +71,52 @@ def test_fsx_lustre(
     _test_data_repository_task(remote_command_executor, mount_dir, bucket_name, fsx_fs_id, region)
 
 
+@pytest.mark.parametrize("deployment_type, per_unit_storage_throughput", [("PERSISTENT_1", 200)])
```
Comment (tilne) [inline tests/integration-tests/tests/storage/test_fsx_lustre.py:74]: Can you hardcode `deployment_type` and `per_unit_storage_throughput` in the config files instead of parameterizing them, since they aren't needed by the validation logic?

File: tests/integration-tests/tests/storage/test_fsx_lustre/test_fsx_lustre_backup/pcluster.config.ini [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+aws_region_name = {{ region }}
+
+[cluster default]
+base_os = {{ os }}
+key_name = {{ key_name }}
+vpc_settings = parallelcluster-vpc
+scheduler = {{ scheduler }}
+master_instance_type = {{ instance }}
+compute_instance_type = {{ instance }}
+{% if scheduler == "awsbatch" %}
```
Comment (tilne) [inline tests/integration-tests/tests/storage/test_fsx_lustre/test_fsx_lustre_backup/pcluster.config.ini:14]: I would drop this `if` statement since these tests only run `sge`.

File: tests/integration-tests/tests/storage/test_fsx_lustre/test_fsx_lustre_backup/pcluster_restore_fsx.config.ini [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+aws_region_name = {{ region }}
+
+[cluster default]
+base_os = {{ os }}
+key_name = {{ key_name }}
+vpc_settings = parallelcluster-vpc
+scheduler = {{ scheduler }}
+master_instance_type = {{ instance }}
+compute_instance_type = {{ instance }}
+{% if scheduler == "awsbatch" %}
```
Comment (tilne) [inline tests/integration-tests/tests/storage/test_fsx_lustre/test_fsx_lustre_backup/pcluster_restore_fsx.config.ini:14]: Same here: I would drop this `if` statement since these tests only run sge.

File: tests/integration-tests/tests/storage/test_fsx_lustre/test_fsx_lustre_backup/pcluster.config.ini [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+[fsx fsx]
+shared_dir = {{ mount_dir }}
+storage_capacity = 1200
+weekly_maintenance_start_time = 1:00:00
+deployment_type = {{ deployment_type }}
+{% if per_unit_storage_throughput %}
+per_unit_storage_throughput = {{ per_unit_storage_throughput }}
+{% endif %}
+automatic_backup_retention_days = 2
```
Comment (tilne) [inline tests/integration-tests/tests/storage/test_fsx_lustre/test_fsx_lustre_backup/pcluster.config.ini:37]: In my opinion, `deployment_type` and `per_unit_storage_throughput` should be hardcoded since the test doesn't validate anything based on these values.

File: tests/integration-tests/tests/storage/test_fsx_lustre.py [hunk 5]
```diff
@@ -71,6 +71,52 @@ def test_fsx_lustre(
...
+):
+    """
+    Test all FSx Lustre related features.
+
+    Grouped all tests in a single function so that cluster can be reused for all of them.
+    """
+    mount_dir = "/fsx_mount_dir"
+    cluster_config = pcluster_config_reader(
+        mount_dir=mount_dir, deployment_type=deployment_type, per_unit_storage_throughput=per_unit_storage_throughput,
+    )
```
Comment (tilne) [inline tests/integration-tests/tests/storage/test_fsx_lustre.py:96]: In the first cluster, can you set a `daily_automatic_backup_start_time` and then add a check that verifies a backup is created automatically, and that tags are copied to it?

File: tests/integration-tests/tests/storage/test_fsx_lustre.py [hunk 6]
```diff
@@ -198,3 +244,45 @@ def _test_data_repository_task(remote_command_executor, mount_dir, bucket_name,
...
+    remote_command_executor.run_remote_command(
+        "echo 'FSx Lustre Backup test file' > {mount_dir}/file_to_backup".format(mount_dir=mount_dir)
+    )
+    job_command = "cat {mount_dir}/file_to_backup ".format(mount_dir=mount_dir)
+    result = sge_commands.submit_command(job_command)
+    job_id = sge_commands.assert_job_submitted(result.stdout)
+    sge_commands.wait_job_completed(job_id)
+    sge_commands.assert_job_succeeded(job_id)
+    result = remote_command_executor.run_remote_command("cat {mount_dir}/file_to_backup".format(mount_dir=mount_dir))
+    assert_that(result.stdout).is_equal_to("FSx Lustre Backup test file")
```
Comment (tilne) [inline tests/integration-tests/tests/storage/test_fsx_lustre.py:261]: What's the purpose of the last two lines?

Comment (siddharthsalot) [reply tests/integration-tests/tests/storage/test_fsx_lustre.py:268]: It is just to assert that the file exists after the sge job is completed. This check ensures that the original cluster had this file in the shared_dir with the content `FSx Lustre Backup test file`.

File: tests/integration-tests/tests/storage/test_fsx_lustre.py [hunk 7]
```diff
@@ -71,6 +72,58 @@ def test_fsx_lustre(
...
+@pytest.mark.skip_dimensions("*", "m6g.xlarge", "centos7", "*")
+@pytest.mark.skip_dimensions("*", "m6g.xlarge", "ubuntu1604", "*")
+def test_fsx_lustre_backup(
+    region, pcluster_config_reader, clusters_factory, test_datadir, os,
+):
+    """
+    Test all FSx Lustre related features.
+
+    Grouped all tests in a single function so that cluster can be reused for all of them.
+    """
```
Comment (tilne) [inline tests/integration-tests/tests/storage/test_fsx_lustre.py:92]: Can you update this docstring? In it, can you summarize all of the functionality being tested (automatic backup time, backups not deleted, backup contains test file which is present when restoring  a file system from it)?

File: tests/integration-tests/tests/storage/test_fsx_lustre.py [hunk 8]
```diff
@@ -71,6 +72,58 @@ def test_fsx_lustre(
...
+    mount_dir = "/fsx_mount_dir"
+    utc_now_plus_15 = datetime.datetime.utcnow() + datetime.timedelta(minutes=15)
+    daily_automatic_backup_start_time = utc_now_plus_15.strftime("%H:%M")
+    logging.info("daily_automatic_backup_start_time" + daily_automatic_backup_start_time)
+    cluster_config = pcluster_config_reader(
+        mount_dir=mount_dir, daily_automatic_backup_start_time=daily_automatic_backup_start_time
+    )
+    cluster = clusters_factory(cluster_config)
```
Comment (tilne) [inline tests/integration-tests/tests/storage/test_fsx_lustre.py:100]: Since there's a lot going on in this function, can you add comments before each cluster creation that explain what portion of the overall testing is fulfilled by it?
